main/commons/src/main/java/org/cryptomator/common/settings/VaultSettings.java

```diff
@@ -138,7 +136,10 @@ public StringProperty customMountPath() {
 
 	public Optional<String> getCustomMountPath() {
 		if (useCustomMountPath.get()) {
-			return Optional.ofNullable(Strings.emptyToNull(customMountPath.get()));
+			String mountPath = Strings.emptyToNull(customMountPath.get());
+			assert mountPath != null; //There must be a valid mountpath we are supposed to use it
```

Comment:
While the new code may prevent this state, we can not assert this when loading the vault settings from disk. Some users may run into NPEs when updating to this change.

Reply:
Correct. That's the second scenario outlined [here.](https://github.com/cryptomator/cryptomator/issues/1547#issuecomment-782356373)
This is part of this PR because for the moment I'm not sure on how to fix this. Should the user be prompted with a message that indicates a config conflict? Should the config be changed silently?

Reply:
This needs to be fixed in the UI, not in the config. You can do assertions for states during runtime, but not persistent configs. We need a resilient approach here, i.e. `null` must be allowed. The fix needs to be done in the UI only.

Reply:
See #1609